openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -286,7 +286,7 @@ def get(self, request, lib_key_str):
         """
         key = LibraryLocatorV2.from_string(lib_key_str)
         api.require_permission_for_library_key(key, request.user, permissions.CAN_VIEW_THIS_CONTENT_LIBRARY)
-        result = api.get_library(key)
+        result = api.get_library(key, user=request.user)
         return Response(ContentLibraryMetadataSerializer(result).data)
```

Comment:
As mentioned, I would prefer that we go with something like this:

```python
result = api.get_library(key)
serializer = ContentLibraryMetadataSerializer(result, context={'request': self.request})
return Response(serializer.data)
```

This is the convention for how the serializer can get access to the request, and this maintains the existing isolation of user-specific/permissions code in the REST API only.

Reply:
Thanks for the suggestion! Updated: https://github.com/openedx/edx-platform/pull/34934/commits/c303069324b444647ac3006fee882ba42d7f42e7#diff-aee1ed7cd71a9cbd5d28d029e3589f4391e7ecc0259178a20a48cbb4f752aea5R285